### ISSUE TITLE
ipv6 support?

### ISSUE THREAD

**Issue by afriggeri** [jupyter-incubator/enterprise_gateway#428]
When trying to run on a machine connected to the network through ipv6 only, the application crashes when importing `enterprise_gateway.processproxies.processproxy` inside `get_local_ip`:

```
  File "[...]/enterprise_gateway/__init__.py", line 4, in <module>
    from .enterprisegatewayapp import launch_instance
  File "[...]/enterprise_gateway/enterprisegatewayapp.py", line 30, in <module>
    from .services.kernels.remotemanager import RemoteMappingKernelManager
  File "[...]/enterprise_gateway/services/kernels/remotemanager.py", line 11, in <module>
    from ..processproxies.processproxy import LocalProcessProxy, RemoteProcessProxy
  File "[...]/enterprise_gateway/services/processproxies/processproxy.py", line 78, in <module>
    local_ip = get_local_ip()
  File "[...]/enterprise_gateway/services/processproxies/processproxy.py", line 76, in get_local_ip
    return localinterfaces.public_ips()[0]  # all were blacklisted, so go with the first
IndexError: list index out of range
```

Are there any plans to support ipv6 in the future?

**Comment by kevin-bates** [comment]
@afriggeri - thanks for the issue.

I would like to say 'yes' but we're reliant on the underlying jupyter framework for this since that's where `localinterfaces` is derived - among other things.  My initial searches for whether jupyter supports ipv6 (including the sources we rely on) are not finding any meaningful hits.  Do you happen to know where jupyter stands on this?

**Comment by afriggeri** [comment]
Afaik, Jupyter runs fine on machines only accessible through ipv6 (because that's where we deploy it).
A middle ground solution could be to move the definition of `local_ip` here: https://github.com/jupyter-incubator/enterprise_gateway/blob/master/enterprise_gateway/services/processproxies/processproxy.py#L79 to a method on BaseProcessProxyABC/RemoteProxy, that way we could at least subclass it and overwrite it with our own ipv6 implementation?

**Comment by kevin-bates** [comment]
@afriggeri - that's a good idea.  I'm thinking on the abstract class is probably fine.  Still not sure if this is going to affect the communication port stuff, tunneling, etc.  Haven't really dealt with ipv6 much.

Do you mind taking a look at this?

**Comment by MR-GOYAL** [comment]
@kevin-bates Any update on this ? I am using version 3.2.2 and still facing same error when running with IPV6.

**Comment by kevin-bates** [comment]
No. This will likely need to be contributed by someone outside the maintainer space. I don't have an environment (or time) to look at this. Sorry.
